measure.go

```diff
@@ -1,54 +1,55 @@
 // Package measure provides a Datastore wrapper that records metrics
-// using github.com/codahale/metrics.
+// using github.com/ipfs/go-metrics-interface
 package measure
 
 import (
 	"io"
 	"time"
 
-	"github.com/codahale/metrics"
 	"github.com/ipfs/go-datastore"
 	"github.com/ipfs/go-datastore/query"
+	"github.com/ipfs/go-metrics-interface"
 )
 
-// Histogram measurements exceeding these limits are dropped. TODO
-// maybe it would be better to cap the value? Should we keep track of
-// drops?
-const (
-	maxLatency = int64(1 * time.Second)
-	maxSize    = int64(1 << 32)
+var (
+	// sort latencies in buckets with following upper bounds in microseconds
+	datastoreLatencyBuckets = []float64{1e2, 1e3, 1e4, 1e5}
 )
 
 // New wraps the datastore, providing metrics on the operations. The
 // metrics are registered with names starting with prefix and a dot.
-//
-// If prefix is not unique, New will panic. Call Close to release the
-// prefix.
 func New(prefix string, ds datastore.Datastore) *measure {
 	m := &measure{
 		backend: ds,
 
-		putNum:     metrics.Counter(prefix + ".Put.num"),
-		putErr:     metrics.Counter(prefix + ".Put.err"),
-		putLatency: metrics.NewHistogram(prefix+".Put.latency", 0, maxLatency, 3),
-		putSize:    metrics.NewHistogram(prefix+".Put.size", 0, maxSize, 3),
-
-		getNum:     metrics.Counter(prefix + ".Get.num"),
-		getErr:     metrics.Counter(prefix + ".Get.err"),
-		getLatency: metrics.NewHistogram(prefix+".Get.latency", 0, maxLatency, 3),
-		getSize:    metrics.NewHistogram(prefix+".Get.size", 0, maxSize, 3),
-
-		hasNum:     metrics.Counter(prefix + ".Has.num"),
-		hasErr:     metrics.Counter(prefix + ".Has.err"),
-		hasLatency: metrics.NewHistogram(prefix+".Has.latency", 0, maxLatency, 3),
-
-		deleteNum:     metrics.Counter(prefix + ".Delete.num"),
-		deleteErr:     metrics.Counter(prefix + ".Delete.err"),
-		deleteLatency: metrics.NewHistogram(prefix+".Delete.latency", 0, maxLatency, 3),
-
-		queryNum:     metrics.Counter(prefix + ".Query.num"),
-		queryErr:     metrics.Counter(prefix + ".Query.err"),
-		queryLatency: metrics.NewHistogram(prefix+".Query.latency", 0, maxLatency, 3),
+		putNum: metrics.New(prefix+".put.num", "Total number of Datastore.Put calls").Counter(),
+		putErr: metrics.New(prefix+".put.err", "Number of errored Datastore.Put calls").Counter(),
+		putLatency: metrics.New(prefix+".put.latency",
+			"Latency distribution of Datastore.Put calls").Histogram(datastoreLatencyBuckets),
+		putSize: metrics.New(prefix+".put.size",
+			"Stored []byte length distribution").Histogram(datastoreLatencyBuckets),
```

Comment:
You are using the same buckets for latency and size, I would suggest buckets like `1<<6, 1<<12, 1<<18, 1<<24` or similar for the size.